### PR TITLE
Reworking the OTel overview to take advantage of the Aspire Dashboard

### PR DESCRIPTION
**Pull request by samsp-msft**
## Summary

Reworking the observability docs to include Aspire, which became too big for a single topic, so its now broken up.
Update to the TOC to show the sub topics.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/diagnostics/observability-applicationinsights.md](https://github.com/dotnet/docs/blob/7f2c2221cca0956e9325d1e34a0a1da5589224e6/docs/core/diagnostics/observability-applicationinsights.md) | [docs/core/diagnostics/observability-applicationinsights](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/observability-applicationinsights?branch=pr-en-us-42304) |
| [docs/core/diagnostics/observability-OTLP-example.md](https://github.com/dotnet/docs/blob/7f2c2221cca0956e9325d1e34a0a1da5589224e6/docs/core/diagnostics/observability-OTLP-example.md) | [docs/core/diagnostics/observability-OTLP-example](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/observability-OTLP-example?branch=pr-en-us-42304) |
| [docs/core/diagnostics/observability-PrGrJa-example.md](https://github.com/dotnet/docs/blob/7f2c2221cca0956e9325d1e34a0a1da5589224e6/docs/core/diagnostics/observability-PrGrJa-example.md) | [docs/core/diagnostics/observability-PrGrJa-example](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/observability-PrGrJa-example?branch=pr-en-us-42304) |
| [docs/core/diagnostics/observability-with-otel.md](https://github.com/dotnet/docs/blob/7f2c2221cca0956e9325d1e34a0a1da5589224e6/docs/core/diagnostics/observability-with-otel.md) | [docs/core/diagnostics/observability-with-otel](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/observability-with-otel?branch=pr-en-us-42304) |

### CODE REVIEW

File: docs/core/diagnostics/observability-OTLP-example.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+In addition to being a standard part of .NET Aspire, the Aspire Dashboard is available as a [standalone docker container](/dotnet/aspire/fundamentals/dashboard/standalone?tabs=powershell), which provides an OTLP endpoint telemetry can be sent to, and it will visualize the logs, metrics and traces. Using the dashboard in this way has no dependency on Aspire, it will visualize telemetry from any application sending it telemetry via OTLP. It works equally well for applications written in Java, GoLang, Python etc. provided that they can send their telemetry to an OTLP endpoint.
+
+Using the Aspire Dashboard has less configuration and moving pieces than using Open Source solutions such as [Prometheus, Grafana and Jaeger](./observability-PrGrJa-example.md), but unlike those tools, the Aspire Dashboard is intended as a developer visualization tool, and not for production monitoring.
+
+## 1. Create the project
+
+Create a simple web API project by using the **ASP.NET Core Empty** template in Visual Studio or the following .NET CLI command:
+
+``` shell
```
Comment (IEvangelist) [inline docs/core/diagnostics/observability-OTLP-example.md:20]:
```suggestion
``` dotnetcli
```

File: docs/core/diagnostics/observability-PrGrJa-example.md [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+# Example: Use OpenTelemetry with Prometheus, Grafana, and Jaeger
+
+This example uses Prometheus for metrics collection, Grafana for creating a dashboard, and Jaeger to show distributed tracing.
+
+## 1. Create the project
+
+Create a simple web API project by using the **ASP.NET Core Empty** template in Visual Studio or the following .NET CLI command:
+
+``` shell
```
Comment (IEvangelist) [inline docs/core/diagnostics/observability-PrGrJa-example.md:16]:
```suggestion
``` dotnetcli
```

File: docs/core/diagnostics/observability-applicationinsights.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+Take the same project from [Step 5](./observability-OTLP-example.md#5-configure-opentelemetry-with-the-correct-providers) and add the following NuGet package:
+
+``` xml
+<ItemGroup>
+  <PackageReference Include="Azure.Monitor.OpenTelemetry.AspNetCore" Version="1.3.0-beta.1" />
+</ItemGroup>
+```
+
+> [!Note]
```
Comment (IEvangelist) [inline docs/core/diagnostics/observability-applicationinsights.md:26]:
```suggestion
> [!NOTE]
```

File: docs/core/diagnostics/observability-applicationinsights.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+## 3. Specify the connection string
+
+If you're not already an Azure customer, you can create a free account at [https://azure.microsoft.com/free/](https://azure.microsoft.com/free/). Log in to the Azure Portal, and either select an existing Application Insights resource or create a new one with [https://ms.portal.azure.com/#create/Microsoft.AppInsights](https://ms.portal.azure.com/#create/Microsoft.AppInsights).
+
+Application Insights identifies which instance to use to store and process data through an instrumentation key and connection string that are found at the top right side of the portal UI.
+
+[![Connection String in Azure Portal](./media/portal_ui.thumb.png)](./media/portal_ui.png#lightbox)
+
+If you're using Azure App Service, this connection string is automatically passed to the application as an environment variable. For other services or when running locally, you need to pass it using the `APPLICATIONINSIGHTS_CONNECTION_STRING` environment variable or in `appsettings.json`. For running locally, it's easiest to add the value to `appsettings.development.json`:
```
Comment (IEvangelist) [inline docs/core/diagnostics/observability-applicationinsights.md:47]:
```suggestion
If you're using Azure App Service, this connection string is automatically passed to the application as an environment variable. For other services or when running locally, you need to pass it using the `APPLICATIONINSIGHTS_CONNECTION_STRING` environment variable or in _appsettings.json_. For running locally, it's easiest to add the value to _appsettings.development.json_:
```

File: docs/core/diagnostics/observability-applicationinsights.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+```json
+"AzureMonitor": {
+    "ConnectionString": "InstrumentationKey=12345678-abcd-abcd-abcd-12345678..."
+}
+```
+
+> [!Note]
```
Comment (IEvangelist) [inline docs/core/diagnostics/observability-applicationinsights.md:55]:
```suggestion
> [!NOTE]
```

File: docs/core/diagnostics/observability-with-otel.md [hunk 6]
```diff
@@ -84,361 +84,53 @@ The following table describes the main packages.
...
-Start Prometheus, and look in the output for the port it's running on, typically 9090:
-
-``` shell
->prometheus.exe
-...
-ts=2023-06-16T05:29:02.789Z caller=web.go:562 level=info component=web msg="Start listening for connections" address=0.0.0.0:9090
-```
-
-Open this URL in your browser. In the Prometheus UI you should now be able to query for your metrics. Use the highlighted button in the following image to open the metrics explorer, which shows all the available metrics.
+[Aspire](/dotnet/aspire/get-started/aspire-overview) is a set of extensions to .NET to make it easy to create and work with distributed applications. One of the benefits of using Aspire is that telemetry is built in, using the OpenTelemetry libraries for .NET. The default project templates for Aspire contain a `ServiceDefaults` project, part of which is to setup and configure OTel. The Service Defaults project is referenced and initialized by each service in an Aspire solution.
```
Comment (IEvangelist) [inline docs/core/diagnostics/observability-with-otel.md:97]: In docs, it has to always be ".NET Aspire" per marketing.

```suggestion
[.NET Aspire](/dotnet/aspire/get-started/aspire-overview) is a set of extensions to .NET to make it easy to create and work with distributed applications. One of the benefits of using .NET Aspire is that telemetry is built in, using the OpenTelemetry libraries for .NET. The default project templates for Aspire contain a `ServiceDefaults` project, part of which is to setup and configure OTel. The Service Defaults project is referenced and initialized by each service in a .NET Aspire solution.
```

File: docs/core/diagnostics/observability-with-otel.md [hunk 7]
```diff
@@ -84,361 +84,53 @@ The following table describes the main packages.
...
+### Reusing Service Defaults project without Aspire Orchestration
 
-From the hamburger menu choose connections, and then enter the text `prometheus` to select your endpoint type. Select **Create a Prometheus data source** to add a new data source.
+Probably the easiest way to configure OTel for ASP.NET projects is to use the Aspire Service Defaults project, even if not using the rest of Aspire such as the AppHost for orchestration. The Service Defaults project is available as a project template via Visual Studio or `dotnet new`. It configures OTel and sets up the OTLP exporter. You can then use the [OTel environment variables](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol#exporter-configuration) to configure the OTLP endpoint to send telemetry to, and provide the resource properties for the application.
 
-[![Grafana connection to prometheus](./media/grafana-connections.thumb.png)](./media/grafana-connections.png#lightbox)
+The steps to use ServiceDefaults outside Aspire are:
```
Comment (IEvangelist) [inline docs/core/diagnostics/observability-with-otel.md:115]:
```suggestion
The steps to use _ServiceDefaults_ outside .NET Aspire are:
```

File: docs/core/diagnostics/observability-with-otel.md [hunk 8]
```diff
@@ -84,361 +84,53 @@ The following table describes the main packages.
...
-[![App Insights logs view](./media/azure-logs.thumb.png)](./media/azure-logs.png#lightbox)
-   :::column-end:::
-   :::column span="":::
-   **Metrics**
-
-[![App Insights metrics view](./media/azure-metrics-graph.thumb.png)](./media/azure-metrics-graph.png#lightbox)
-   :::column-end:::
-   :::column span="":::
-   **Distributed Tracing**
+Note: service Defaults can setup the following additional functionality if required via `AddServiceDefaults()` or the specific functions:
```
Comment (noahfalk) [inline docs/core/diagnostics/observability-with-otel.md:132]:
```suggestion
Note: Service defaults can setup the following additional functionality if required via `AddServiceDefaults()` or the specific functions:
```

File: docs/core/diagnostics/observability-with-otel.md [hunk 9]
```diff
@@ -84,361 +84,53 @@ The following table describes the main packages.
...
 
-[![App Insights transaction view](./media/azure-tracing.thumb.png)](./media/azure-tracing.png#lightbox)
-   :::column-end:::
-:::row-end:::
+- Health checks with `/health` and `/alive` endpoints
+- Service discovery which will be a no-op without the rest of Aspire
```
Comment (noahfalk) [inline docs/core/diagnostics/observability-with-otel.md:135]:
```suggestion
- Service discovery which will be a no-op without the rest of .NET Aspire
```

File: docs/core/diagnostics/observability-with-otel.md [hunk 10]
```diff
@@ -84,361 +84,53 @@ The following table describes the main packages.
...
+- Configuring reslience for HttpClient which will retry the request in the case of failures
```
Comment (noahfalk) [inline docs/core/diagnostics/observability-with-otel.md:136]:
```suggestion
- Configuring resilience for HttpClient which will retry the request in the case of failures
```

File: docs/core/diagnostics/snippets/OTLP-Example/csharp/OTLP-Example.csproj [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <RootNamespace>OTLP_Example</RootNamespace>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DefineConstants>$(DefineConstants);AZURE_MONITOR</DefineConstants>
```
Comment (noahfalk) [inline docs/core/diagnostics/snippets/OTLP-Example/csharp/OTLP-Example.csproj:11]: Does this AZURE_MONITOR define matter for something? It looked like it was unused.

**Review by noahfalk — APPROVED** [review]
Minor comments inline but LGTM!

File: docs/core/diagnostics/observability-applicationinsights.md (hunk 3 above)
Comment (samsp-msft) [reply docs/core/diagnostics/observability-applicationinsights.md]: Elsewhere its using the mixed case, should it be all caps everywhere?

Comment (IEvangelist) [reply docs/core/diagnostics/observability-applicationinsights.md]: Yes, it should be all caps everywhere.

**Pull request merged**